R/exclude_nca_extra.R

```diff
@@ -0,0 +1,91 @@
+# This file will only be used until PKNCA includes the additional functions in their new version
+# ToDo(Gerardo): Remove the file once PKNCA incorporates all our functions (adj.r.squared...)
+
+
+#' Exclude NCA Results Based on Parameter Thresholds
+#'
+#' Exclude rows from NCA results based on specified thresholds for a given parameter.
+#' This function allows users to define minimum and/or maximum acceptable values
+#' for a parameter and excludes rows that fall outside these thresholds.
+#'
+#' @param parameter The name of the PKNCA parameter to evaluate (e.g., "span.ratio").
+#' @param min_thr The minimum acceptable value for the parameter. If not provided, is not applied.
+#' @param max_thr The maximum acceptable value for the parameter. If not provided, is not applied.
+#' @param affected_parameters Character vector of PKNCA parameters that will be marked as excluded.
+#'                            By default is the defined parameter.
+#' @returns A function that can be used with `PKNCA::exclude` to mark through the 'exclude'  column
+#'          the rows in the PKNCA results based on the specified thresholds for a parameter.
+#' @examples
+#' # Example dataset
+#' my_data <- PKNCA::PKNCAdata(
+#'   PKNCA::PKNCAconc(data.frame(conc = c(1, 2, 3, 4),
+#'                               time = c(0, 1, 2, 3),
+#'                               subject = 1),
+#'                    conc ~ time | subject),
+#'   PKNCA::PKNCAdose(data.frame(subject = 1, dose = 100, time = 0),
+#'                    dose ~ time | subject)
+#' )
+#' my_result <- PKNCA::pk.nca(my_data)
+#'
+#' # Exclude rows where span.ratio is less than 100
+#' excluded_result <- PKNCA::exclude(
+#'   my_result,
+#'   FUN = exclude_nca_by_param("span.ratio", min_thr = 100)
+#' )
+#' as.data.frame(excluded_result)
+#'
+#' @export
+
+exclude_nca_by_param <- function(parameter, min_thr = NULL, max_thr = NULL, affected_parameters) {
+
+  # Determine if thresholds are defined and if so check they are single numeric objects
+  thr_def <- validate_thresholds(min_thr, max_thr)
+  if (missing(affected_parameters)) affected_parameters <- parameter
```

Comment:
issue: Please do not include arguments without defaults after arguments with defaults. Also, there is no need to use `missing()` like that - you can assign function arguments to other arguments:
```suggestion
exclude_nca_by_param <- function(parameter, min_thr = NULL, max_thr = NULL, affected_parameters = parameter) {

  # Determine if thresholds are defined and if so check they are single numeric objects
  thr_def <- validate_thresholds(min_thr, max_thr)
```